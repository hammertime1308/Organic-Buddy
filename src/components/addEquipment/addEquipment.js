import React, { useState } from 'react';
import {
  ActivityIndicator,
  Button,
  Image,
  Modal,
  ScrollView,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
import { RadioButton } from 'react-native-paper';
import { launchImageLibrary } from 'react-native-image-picker';
import styled from 'styled-components/native';

import { addNewEquipment } from '../../api';

import { COLOR } from '../../styles';

const ModalContainer = styled.View`
  flex: 1;
  background-color: rgba(0, 0, 0, 0.5);
  justify-content: center;
  align-items: center;
`;

const ModalView = styled.View`
  background-color: ${COLOR.White};
  width: 90%;
  height: 90%;
  border-radius: 5px;
  align-items: center;
`;

export const AddEquipmentModal = ({ visible, user, closeModal }) => {
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [price, setPrice] = useState('');
  const [image, setImage] = useState('');
  const [checked, setChecked] = useState('Sell');

  const [loading, setLoading] = useState(false);

  const clearData = () => {
    setName('');
    setDescription('');
    setPrice('');
    setImage('');
    setChecked('Sell');
  };

  const isValid = (name, description, price) => {
    if (name === '') {
      alert('Add equipment name');
      return false;
    } else if (description === '') {
      alert('Add some description');
      return false;
    } else if (price === '') {
      alert('Add price');
      return false;
    }
    return true;
  };

  const selectImages = async () => {
    let options = {
      title: 'Select Image',
      mediaType: 'photo',
      quality: 0.3,
      includeBase64: true,
      storageOptions: {
        skipBackup: true,
        path: 'images',
      },
    };
    launchImageLibrary(options, response => {
      if (response.didCancel) {
        setImage('');
      } else if (response.error) {
        alert('Error: ', response.error);
      } else {
        setImage(response.base64);
      }
    });
  };

  const apiCall = async () => {
    if (isValid(name, description, price)) {
      setLoading(true);
      let response = await addNewEquipment({
        equipmentName: name,
        equipmentDescription: description,
        equipmentPrice: price,
        equipmentImages: image === '' ? [] : [image],
        equipmentType: checked,
        sellerId: user.userId,
        sellerContact: user.contact,
      });
      setLoading(false);
      if (response.status === 200) {
        clearData();
        closeModal();
      } else {
        alert(response.data);
      }
    }
  };

  return (
    <Modal
      visible={visible}
      transparent
      onRequestClose={() => {
        clearData();
        closeModal();
      }}
      animationType="fade">
      <ModalContainer>
        <ModalView>
          <Text
            style={{
              fontSize: 20,
              fontWeight: 'bold',
              color: `${COLOR.GreenHeader}`,
              paddingTop: 20,
            }}>
            Advertise for equipment
          </Text>
          {loading ? (
            <ActivityIndicator
              style={{ position: 'absolute', top: '50%' }}
              size={50}
              color={COLOR.GreenHeader}
            />
          ) : (
            <ScrollView
              style={{
                flex: 1,
                width: '100%',
                padding: 20,
              }}>
              <TextInput
                placeholder="Name of equipment"
                value={name}
                onChangeText={setName}
                style={{
                  borderColor: 'black',
                  borderWidth: 0.2,
                  textAlign: 'center',
                  marginTop: 20,
                }}
              />
              <TextInput
                placeholder="Description of equipment"
                value={description}
                onChangeText={setDescription}
                style={{
                  borderColor: 'black',
                  borderWidth: 0.2,
                  textAlign: 'center',
                  marginTop: 20,
                }}
              />
              <TextInput
                placeholder="Price"
                value={price}
                keyboardType="numeric"
                onChangeText={setPrice}
                style={{
                  borderColor: 'black',
                  borderWidth: 0.2,
                  textAlign: 'center',
                  marginTop: 20,
                }}
              />
              <View
                style={{
                  flexDirection: 'row',
                  justifyContent: 'space-evenly',
                  alignItems: 'center',
                  marginTop: 20,
                }}>
                <Text style={{ fontSize: 17, fontWeight: 'bold' }}>Sell</Text>
                <RadioButton
                  value="Sell"
                  status={checked === 'Sell' ? 'checked' : 'unchecked'}
                  onPress={() => setChecked('Sell')}
                />
              </View>
              <View
                style={{
                  flexDirection: 'row',
                  justifyContent: 'space-evenly',
                  alignItems: 'center',
                }}>
                <Text style={{ fontSize: 17, fontWeight: 'bold' }}>Hire</Text>
                <RadioButton
                  value="Hire"
                  status={checked === 'Hire' ? 'checked' : 'unchecked'}
                  onPress={() => setChecked('Hire')}
                />
              </View>

              <Image
                style={{
                  width: 100,
                  height: 100,
                  marginBottom: 20,
                  alignSelf: 'center',
                }}
                source={{
                  uri: 'data:image/jpeg;base64,' + image,
                }}
              />

              <Button
                title="add image"
                color={COLOR.GreenHeader}
                onPress={selectImages}
              />
            </ScrollView>
          )}
          <TouchableOpacity
            activeOpacity={0.8}
            onPress={apiCall}
            style={{
              backgroundColor: `${COLOR.GreenHeader}`,
              position: 'absolute',
              bottom: 0,
              width: '100%',
              height: '10%',
              justifyContent: 'center',
              alignItems: 'center',
              borderBottomLeftRadius: 5,
              borderBottomRightRadius: 5,
            }}>
            <Text style={{ fontSize: 20, color: 'white', fontWeight: 'bold' }}>
              Submit
            </Text>
          </TouchableOpacity>
        </ModalView>
      </ModalContainer>
    </Modal>
  );
};
