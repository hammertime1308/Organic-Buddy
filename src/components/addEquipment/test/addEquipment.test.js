import React from 'react';
import renderer from 'react-test-renderer';
import { AddEquipmentModal } from '../addEquipment';

describe('Add Equipment', () => {
  it('Should not render Modal', () => {
    const tree = renderer
      .create(
        <AddEquipmentModal
          visible={false}
          user={{ userId: '0' }}
          closeModal={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render modal', () => {
    const tree = renderer
      .create(
        <AddEquipmentModal
          visible={true}
          user={{ userId: '0' }}
          closeModal={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
