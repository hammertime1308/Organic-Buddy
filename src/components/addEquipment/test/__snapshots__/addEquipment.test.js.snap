// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Equipment Should not render Modal 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffff",
            "borderRadius": 5,
            "height": "90%",
            "width": "90%",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(76, 154, 42, 1)",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingTop": 20,
          }
        }
      >
        Advertise for equipment
      </Text>
      <RCTScrollView
        style={
          Object {
            "flex": 1,
            "padding": 20,
            "width": "100%",
          }
        }
      >
        <View>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Name of equipment"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 0.2,
                "marginTop": 20,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Description of equipment"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 0.2,
                "marginTop": 20,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Price"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 0.2,
                "marginTop": 20,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-evenly",
                "marginTop": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
            >
              Sell
            </Text>
            <View
              accessibilityLiveRegion="polite"
              accessibilityRole="radio"
              accessibilityState={
                Object {
                  "checked": true,
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                    "padding": 6,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#03dac4",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰄬
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
            >
              Hire
            </Text>
            <View
              accessibilityLiveRegion="polite"
              accessibilityRole="radio"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                    "padding": 6,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "opacity": 0,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#03dac4",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰄬
                </Text>
              </View>
            </View>
          </View>
          <Image
            source={
              Object {
                "uri": "data:image/jpeg;base64,",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "marginBottom": 20,
                "width": 100,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "rgba(76, 154, 42, 1)",
                    },
                  ]
                }
              >
                add image
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(76, 154, 42, 1)",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "bottom": 0,
            "height": "10%",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Add Equipment Should render modal 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffff",
            "borderRadius": 5,
            "height": "90%",
            "width": "90%",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(76, 154, 42, 1)",
            "fontSize": 20,
            "fontWeight": "bold",
            "paddingTop": 20,
          }
        }
      >
        Advertise for equipment
      </Text>
      <RCTScrollView
        style={
          Object {
            "flex": 1,
            "padding": 20,
            "width": "100%",
          }
        }
      >
        <View>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Name of equipment"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 0.2,
                "marginTop": 20,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Description of equipment"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 0.2,
                "marginTop": 20,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Price"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 0.2,
                "marginTop": 20,
                "textAlign": "center",
              }
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-evenly",
                "marginTop": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
            >
              Sell
            </Text>
            <View
              accessibilityLiveRegion="polite"
              accessibilityRole="radio"
              accessibilityState={
                Object {
                  "checked": true,
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                    "padding": 6,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#03dac4",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰄬
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
            >
              Hire
            </Text>
            <View
              accessibilityLiveRegion="polite"
              accessibilityRole="radio"
              accessibilityState={
                Object {
                  "checked": false,
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 18,
                    "padding": 6,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "opacity": 0,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#03dac4",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰄬
                </Text>
              </View>
            </View>
          </View>
          <Image
            source={
              Object {
                "uri": "data:image/jpeg;base64,",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "marginBottom": 20,
                "width": 100,
              }
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "rgba(76, 154, 42, 1)",
                    },
                  ]
                }
              >
                add image
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(76, 154, 42, 1)",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "bottom": 0,
            "height": "10%",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
